### PR TITLE
Chore: Evaluate Icon - Typescript

### PR DESCRIPTION
**Pull request by AlexAlexandre**
## Proposed changes
Evaluate the Icon component. 

* Change the `app/lib/Icon` to `app/containers/Icon`.
* Type the icon component to autocomplete the icons name and get a error when the name does not exist. 
* Create a doc to the generate type process.

## Issue(s)	
N/A

## How to test or reproduce
Test the app to check if the icons is still working.

## Screenshots
N/A

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments
N/A

### CODE REVIEW

File: docs/icons.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Icons
+
+Currently we use the [IcoMoon App](https://icomoon.io/app), a free icon library. To use that, we also need the [react-native-vector-icons](https://github.com/oblador/react-native-vector-icons), so we can create an icon component with the function:
```
Comment (diegolmello) [inline docs/icons.md:3]: Let me finish this doc later :)

File: storybook/stories/__snapshots__/List.storyshot [hunk 2]
```diff
@@ -1,6 +1,6 @@
...
..."#0d0e12\\"}],\\"numberOfLines\\":1},\\"children\\":[\\"Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries\\"]},{\\"type\\":\\"Text\\",\\"props\\":{\\"allowFontScaling\\":false,\\"style\\":[{\\"fontSize\\":20},null,{\\"fontFamily\\":\\"custom\\",\\"fontWeight\\":\\"normal\\",\\"fontStyle\\":\\"normal\\"},{}]},\\"children\\":[\\"\\"]}]}]},{\\"type\\":\\"View\\",\\"props\\":{\\"style\\":{\\"paddingLeft\\":12}},\\"children\\":[{\\"type\\":\\"View\\",\\"props\\":{\\"style\\":[{\\"alignItems\\":\\"center\\",\\"justifyContent\\":\\"center\\"},null]},\\"children\\":[{\\"type\\":\\"Text\\",\\"props\\":{\\"allowFontScaling\\":false,\\"style\\":[{\\"fontSize\\":20,\\"color\\":\\"#9ca2a8\\"},null,{\\"fontFamily\\":\\"custom\\",\\"fontWeight\\":\\"normal\\",\\"fontStyle\\":\\"normal\\"},{}]},\\"children\\":[\\"\\"]}]}]}]}]},{\\"type\\":\\"View\\",\\"props\\":{\\"style\\":[{\\"height\\":0.5},null,{\\"backgroundColor\\":\\"#cbcbcc\\"}]},\\"children\\":null}]}]}"`;
```
Comment (diegolmello) [inline storybook/stories/__snapshots__/List.storyshot:3]: I didn't get why those stories were updated

**Review by diegolmello — CHANGES_REQUESTED** [review]

File: storybook/stories/__snapshots__/List.storyshot (hunk 2 above)
Comment (AlexAlexandre) [reply storybook/stories/__snapshots__/List.storyshot]: was by mistake, removed. Good catch!

File: scripts/build-icons-name.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+iconsName = `${iconsName.slice(0, -2) }\n};\n`;
+
+fs.writeFile('app/containers/CustomIcon/glyphIcoMoon.js', iconsName, 'utf8', function (err) {
+	if (err) {
+		console.log('An error occurred while writing Object to File.');
+		console.log(err);
+		return;
+	}
+
+	console.log('File has been saved.');
```
Comment (diegolmello) [inline scripts/build-icons-name.js:19]:
```suggestion
	console.log('🚀   Icons name generated.');
```

**Review by diegolmello — CHANGES_REQUESTED** [review]
Almost there! 🤞🏼

File: scripts/build-icons-name.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+const fs = require('fs');
+const icoMoonConfig = require('../app/containers/CustomIcon/selection.json');
```
[the remark this answers is not in the document]
Comment (AlexAlexandre) [reply scripts/build-icons-name.js]: As discussed on a meeting, the script only will throw an error if we change the file to another folder. 
We can run from any place/folder by `yarn build-icon-set` and will work.

e.g: I'm on `Rocket.Chat.ReactNative/app/views/RoomActionsView` and works fine. 

Screen Shot 2022-04-04 at 12 26 13

File: app/containers/CustomIcon/index.tsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+const icoMoonConfig = require('./selection.json');
+
+export const IconSet = createIconSetFromIcoMoon(icoMoonConfig, 'custom', 'custom.ttf');
+
+export type TIconsName = keyof typeof mappedIcons;
+
+interface ICustomIcon extends TextProps {
+	name: TIconsName;
+	size: number;
+	color?: string;
```
Comment (diegolmello) [inline app/containers/CustomIcon/index.tsx:16]: It should be required.
With this change, TS is going to complain a little bit with the styles.

Pay attention to:

```tsx
// before
<CustomIcon name='close' style={[styles.close, { color: themes[theme].titleText }]} size={20} />
...
// after: not only add the `color` prop, but remove the array from `style` as well
<CustomIcon name='close' style={styles.close} color={themes[theme].titleText} size={20} />
```

**Review by diegolmello — CHANGES_REQUESTED** [review]

File: app/containers/CustomIcon/index.tsx (hunk 5 above)
Comment (AlexAlexandre) [reply app/containers/CustomIcon/index.tsx]: due this change, I had to update the snapshots removing the colors from the style, can you take a look ?

File: app/views/SelectListView.tsx [hunk 6]
```diff
@@ -151,6 +151,7 @@ class SelectListView extends React.Component<ISelectListViewProps, ISelectListVi
 	};
 
 	renderItem = ({ item }: { item: Partial<IRoom> }) => {
+		console.log('***** enter here *****');
```
Comment (diegolmello) [inline app/views/SelectListView.tsx:154]: Nope

**Review by diegolmello — APPROVED** [review]
After you remove that comment, please trigger the workflow and send this task to QA.

**Pull request merged**